new-log-viewer/src/utils/url.ts

```diff
@@ -1,3 +1,23 @@
+/**
+ * Gets an absolute URL composed of a given path relative to the
+ * window.location, if the given path is a relative reference; otherwise
+ * the given path is returned verbatim.
+ *
+ * @param path The path to be resolved.
+ * @return The absolute URL of the given path.
+ * @throws {Error} if the given `path` is a relative reference but invalid.
```

Comment:
While in Doxygen the `{}` could be omitted, it seems in JSDoc the correct syntax to describe the type of the object to be thrown is with the `{}`: https://jsdoc.app/tags-throws 
(Maybe we can omit the type as well. What do you think?)

I have updated the docs at `LogFileManager.ts` with the correct syntax.

Reply:
Sure, let's do the correct thing and keep `{}`. If we omit the type, are we is relying on typescript to infer it? Or are you saying since it's `Error`, it doesn't matter if we specify the type since it's the base class for errors?

Reply:
Yup, both, with the later reason being stronger. 

I don't have a strong preference though. With those reasons listed, while TypeScript infers throwable types fine, it might be better to document the type of any non-`Error` throwable so readers do not have to look it up in the function body. 

Then for consistency reasons, the `Error` type notations might be worth keeping... 

What do you think?